### PR TITLE
Introducing auto-aim, small tweaks

### PR DESCRIPTION
**Pull request by Daroslaw**
#14

### CODE REVIEW

File: Zion's End/src/Tower.cpp [hunk 1]
```diff
@@ -7,6 +7,7 @@ constexpr uint16 canonTextureWidth = 95;
 constexpr uint16 canonTextureHeight = 160;
 constexpr uint16 towerTextureSize = 256;
 constexpr float DEFAULT_COOLDOWN = 2.f;
+constexpr float DEFAULT_RANGE = 250.f;
```
Comment (Sand3r-) [inline Zion's End/src/Tower.cpp]: Probably shouldn't be duplicated here and in Projectile.cpp

Comment (Sand3r-) [reply Zion's End/src/Tower.cpp]: I think that in general it'd be more professional to store all these constants in files, but idk if we really want to do that

File: Zion's End/src/Tower.cpp [hunk 2]
```diff
@@ -59,14 +62,46 @@ void Tower::Update(sf::Time deltaTime, Resources * resources)
...
+
+			float distance = as::Length(enemy->GetPosition() - m_Position);
+
+			if (distance < minDistance)
+			{
+				minDistance = distance;
+				closestEnemy = enemy;
+			}
+		}
+		if (minDistance <= m_Range && closestEnemy != nullptr)
```
Comment (Sand3r-) [inline Zion's End/src/Tower.cpp]: Not sure if that check for closestEnemy != nullptr is all that necessary.
The function enters this block if the m_Target was nullptr initially anyway.
Performing ``test`` or ``cmp`` takes more cycles than just performing nullpointer (0) write again.

Comment (Sand3r-) [reply Zion's End/src/Tower.cpp]: Sorry if I'm being nitpicky, it's cause of this RE

**Review by Sand3r- — APPROVED** [review]
Good job 👍 I'm pretty sure that this should suffice for now.

File: Zion's End/src/Tower.cpp (hunk 1 above)
Comment (Daroslaw) [reply Zion's End/src/Tower.cpp]: I thought about a possibility of having a tower with smaller sight radius than the range of its projectile, but maybe that doesn't make sense.

File: Zion's End/src/Tower.cpp (hunk 2 above)
Comment (Daroslaw) [reply Zion's End/src/Tower.cpp]: You're right

**Review by Sand3r- — APPROVED** [review]

**Pull request merged**